app/_src/policies/hostnamegenerator.md

```diff
@@ -0,0 +1,70 @@
+---
+title: HostnameGenerator
+---
+
+{% warning %}
+This resource is experimental!
+{% endwarning %}
+
+A `HostnameGenerator` provides
+
+- a template to generate hostnames from properties of `MeshServices`
+- a selector that defines for which `MeshServices` this generator runs
+
+```yaml
+type: HostnameGenerator
+name: all
+spec:
+  selector:
+    meshService:
+      matchLabels:
+        k8s.kuma.io/namespace: kuma-demo
+  template: "{{ .Name }}.{{ .Namespace }}.mesh"
```

Comment:
`.Name` when synced from global will have a hash so it might not be the best for MS. Definitely not good for MES.

Reply:
Discussed and resolved